references/validationFunctions/sample_registration/specimentypeDesignation.js

```diff
@@ -20,9 +23,6 @@ const validation = () => {
             }
         }
         return result;
-    }
-    
-    validate();
-}
+    })();
 
 module.exports = validation;
```

Comment:
module is a node concept, clinical won't be able to run this line if the serialized script going to have this

Reply:
[There is a script ](https://github.com/icgc-argo/argo-dictionary/blob/ws_42/references/script.js), which imports and runs through all the validation functions, grabs only the actual function, removes its "outer layer" (the external function declaration which passes in the variable params), converts it into a string, and inserts it into the dictionary. Clinical will not see the module.exports, or the parameters that are passed in.

For example, here is a snippet of the dictionary after you run `npm compile` and

```
{
    "name": "ICGC-ARGO Data Dictionary",
    "version": "0.0",
    "references": {
        "script": {
            "chemotherapy": {},
            "common": {
                "ajccValidation": "(function validate() {\n\n        let result = {valid: true, message: \"Ok\"};\n        \n        /* This is not a required field, so first ensure that it exists */\n        if ($field){\n            /* Contingent on the naming system for tumour staging systems to remain consistent */\n            const stagingName = $name.trim().toLowerCase().split('_tumour_staging_system')[0];\n            /* Perform validation only if an AJCC value was selected */\n            if (/^(AJCC)\\b/i.test($field)){\n                const requiredFields = [\n                    `${stagingName}_m_category`,\n                    `${stagingName}_n_category`,\n                    `${stagingName}_t_category`\n                ];\n                const convertedRow = Object.fromEntries(Object.entries($row).map(([fieldName,fieldVal]) => [fieldName.toLowerCase(), fieldVal]));\n                /* Check for contigous spaces wrapped with quotes (empty strings) */\n                const checkforEmpty = (entry) => {return /^\\s+$/g.test(decodeURI(entry).replace(/^\"(.*)\"$/, '$1'))};\n                /* search for fields with falsy values, given the field exists */\n                const emptyFields = requiredFields.filter(field => Object.keys(convertedRow).includes(field) && (!convertedRow[field] || checkforEmpty(convertedRow[field])))\n                if (emptyFields.length){\n                    result = {valid: false, \"message\": `The following fields are required when ${$name} is set to an AJCC option: ${emptyFields}`};\n                }\n            }\n        }\n        return result;\n    })()"
            },
            "donor": {
                "ensureDeceased": "(function validate() {\n        let result = {valid: true, message: \"Ok\"};\n        const causeOfDeath = typeof($field) === 'string' ? $field.trim().toLowerCase() : $field;\n        const vitalStatus = typeof($row.vital_status) === 'string' ? $row.vital_status.trim().toLowerCase(): $row.vital_status;\n    \n        if (!causeOfDeath && vitalStatus === \"deceased\"){\n            result = {valid: false, message: \"Cause of death must be provided when the donor's vital_status is deceased.\"}\n        }\n        else if (causeOfDeath && vitalStatus != \"deceased\"){\n            result = {valid: false, message: \"A cause of death cannot be provided if the donor's vital_status is not deceased.\"}\n        }\n        return result;\n    })()"
            },
//etc

```

Reply:
cool, so I assume it was tested against clinical ! 
if yes resolve this thread

Reply:
yep, tested in clinical